# Top Games on Google Play Store - An EDA

From the 'android-games.csv' dataset from [Top Games of Google Play Store](https://www.kaggle.com/dhruvildave/top-play-store-games), we'll answer the following questions using an Exploratory Data Analysis approach:

* What is the most expensive game on this list?
* What is the final star rating for each game?
* What is the average star rating for each category?

### Import the Libraries Used

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### Load and Read Data

In [2]:
data_path = '/kaggle/input/top-play-store-games/android-games.csv'
raw_data = pd.read_csv(data_path)

### Basic Data Information

In [3]:
raw_data.head()

rank                                         title  total ratings installs  \
0     1                  Garena Free Fire - The Cobra       80678661  500.0 M   
1     2                   PUBG MOBILE: Graffiti Prank       35971961  100.0 M   
2     3                     Mobile Legends: Bang Bang       25836869  100.0 M   
3     4                                   Brawl Stars       17181659  100.0 M   
4     5  Sniper 3D: Fun Free Online FPS Shooting Game       14237554  100.0 M   

   average rating  growth (30 days)  growth (60 days)  price     category  \
0            4.33               2.9               7.9    0.0  GAME ACTION   
1            4.24               2.0               3.1    0.0  GAME ACTION   
2            4.08               1.6               3.3    0.0  GAME ACTION   
3            4.27               4.1               6.6    0.0  GAME ACTION   
4            4.33               0.8               1.8    0.0  GAME ACTION   

   5 star ratings  4 star ratings  3 star ratings  2 star ratings  \
0        61935712         4478738         2795172         1814999   
1        26670566         2109631         1352610          893674   
2        17850942         1796761         1066095          725429   
3        12493668         1474319          741410          383478   
4         9657878         2124544         1034025          375159   

   1 star ratings   paid  
0         9654037  False  
1         4945478  False  
2         4397640  False  
3         2088781  False  
4         1045945  False

We then have the following features:
* __rank__: Rank in a particular category
* __title__: Game title
* __total ratings__: Total number of ratings
* __installs__: Approximate install milestone
* __average rating__: Average rating out of 5
* __growth (30 days)__: Percent growth in 30 days
* __growth (60 days)__: Percent growth in 60 days
* __price__: Price in dollars
* __category__: Game category
* __5 star ratings__: Number of 5 star ratings
* __4 star ratings__: Number of 4 star ratings
* __3 star ratings__: Number of 3 star ratings
* __2 star ratings__: Number of 2 star ratings
* __1 star ratings__: Number of 1 star ratings
* __paid__: Whether the game is paid or not

In [4]:
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1730 entries, 0 to 1729
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   rank              1730 non-null   int64  
 1   title             1730 non-null   object 
 2   total ratings     1730 non-null   int64  
 3   installs          1730 non-null   object 
 4   average rating    1730 non-null   float64
 5   growth (30 days)  1730 non-null   float64
 6   growth (60 days)  1730 non-null   float64
 7   price             1730 non-null   float64
 8   category          1730 non-null   object 
 9   5 star ratings    1730 non-null   int64  
 10  4 star ratings    1730 non-null   int64  
 11  3 star ratings    1730 non-null   int64  
 12  2 star ratings    1730 non-null   int64  
 13  1 star ratings    1730 non-null   int64  
 14  paid              1730 non-null   bool   
dtypes: bool(1), float64(4), int64(7), object(3)
memory usage: 191.0+ KB


In [5]:
raw_data.isnull().sum()

rank                0
title               0
total ratings       0
installs            0
average rating      0
growth (30 days)    0
growth (60 days)    0
price               0
category            0
5 star ratings      0
4 star ratings      0
3 star ratings      0
2 star ratings      0
1 star ratings      0
paid                0
dtype: int64

There is no null data and all the columns are filled on every row. This is expected since every game on Play Store must have all these informations and this is a Kaggle dataset.

In [6]:
raw_data.describe()

rank  total ratings  average rating  growth (30 days)  \
count  1730.000000   1.730000e+03     1730.000000       1730.000000   
mean     50.475723   1.101182e+06        4.313410        193.167341   
std      28.941630   3.311738e+06        0.253545       4775.617650   
min       1.000000   3.823800e+04        3.090000          0.000000   
25%      25.000000   1.879988e+05        4.180000          0.100000   
50%      51.000000   4.576750e+05        4.330000          0.500000   
75%      75.750000   9.443342e+05        4.490000          1.600000   
max     100.000000   8.067866e+07        4.910000     140394.400000   

       growth (60 days)        price  5 star ratings  4 star ratings  \
count       1730.000000  1730.000000    1.730000e+03    1.730000e+03   
mean           3.969249     0.012965    7.883837e+05    1.216466e+05   
std           18.912904     0.243492    2.483061e+06    2.955701e+05   
min            0.000000     0.000000    2.189800e+04    2.441000e+03   
25%            0.300000     0.000000    1.358290e+05    2.180200e+04   
50%            1.000000     0.000000    3.109435e+05    5.464350e+04   
75%            3.300000     0.000000    6.511310e+05    1.095648e+05   
max          605.100000     7.490000    6.193571e+07    5.397273e+06   

       3 star ratings  2 star ratings  1 star ratings  
count    1.730000e+03    1.730000e+03    1.730000e+03  
mean     5.954986e+04    2.796247e+04    1.036364e+05  
std      1.447400e+05    7.676625e+04    3.589869e+05  
min      7.070000e+02    2.880000e+02    5.270000e+02  
25%      1.027775e+04    4.529500e+03    1.356075e+04  
50%      2.665850e+04    1.133050e+04    3.569400e+04  
75%      5.581750e+04    2.526650e+04    8.632625e+04  
max      2.795172e+06    1.814999e+06    9.654037e+06

From a preliminary analysis, we can see that more than 75% of the top games are free (by checking when the __price__ is 0.00). In fact, let's check the total:

In [7]:
raw_data['paid'].value_counts(normalize = True) * 100

False    99.595376
True      0.404624
Name: paid, dtype: float64

So only 0.40% of the games in this list are paid games (that acounts for only 7 games).

To find the paid games on this list we use:

In [8]:
raw_data[raw_data['price'] != 0].sort_values(by = 'price', ascending = False)

rank                  title  total ratings installs  average rating  \
211     12              Minecraft        3999232   10.0 M            4.54   
153     54               Terraria         317546    1.0 M            4.65   
477     64                 Reigns         146504  500.0 k            4.58   
261     62          Geometry Dash         767518    1.0 M            4.62   
869     46               Cytus II         109687    1.0 M            4.59   
1588    65  Trivia Crack (No Ads)          74594  500.0 k            4.58   
91      92          Hitman Sniper         848831   10.0 M            4.41   

      growth (30 days)  growth (60 days)  price        category  \
211                1.4               2.8   7.49     GAME ARCADE   
153                1.0               2.3   4.99  GAME ADVENTURE   
477                0.1               0.2   2.99       GAME CARD   
261                0.6               1.2   1.99     GAME ARCADE   
869                1.0               2.2   1.99      GAME MUSIC   
1588               0.9               2.2   1.99     GAME TRIVIA   
91                 0.3               0.8   0.99     GAME ACTION   

      5 star ratings  4 star ratings  3 star ratings  2 star ratings  \
211          3214765          312995          138682           70661   
153           264563           24837           10184            4387   
477           108627           25330            5696            2743   
261           623950           73997           25169            9705   
869            86420           12246            4534            1999   
1588           55411           12625            3123            1117   
91            607502          115334           48040           19224   

      1 star ratings  paid  
211           262126  True  
153            13572  True  
477             4105  True  
261            34694  True  
869             4484  True  
1588            2315  True  
91             58729  True

Here we have _Minecraft_ as the most expensive game on this list, costing $7.49, with more than 10.0 M installs at the time this dataset was gathered.

## Star Ratings

Using the 5 columns that counts the number of different star ratings we can arrive at a final star rating using a weighted average. In this case:

$$\text{final rating} = \frac{5 \times (\text{#5⋆}) + 4 \times (\text{#4⋆}) + 3 \times (\text{#3⋆}) + 2 \times (\text{#2⋆}) + (\text{#1⋆})}{(\text{#5⋆}) +(\text{#4⋆}) +(\text{#3⋆}) +(\text{#2⋆}) +(\text{#1⋆})}$$

where $(\#x⋆)$ indicates the total number of $x$ star ratings for $x = 1, 2, 3, 4, 5$.

In [9]:
final_stars = lambda row: (5*row['5 star ratings'] + 4*row['4 star ratings'] + 3*row['3 star ratings'] \
                           + 2*row['2 star ratings'] + row['1 star ratings']) / (row['5 star ratings'] \
                           + row['4 star ratings'] + row['3 star ratings'] + row['2 star ratings'] \
                           + row['1 star ratings'])
games_with_rating = raw_data
games_with_rating['final rating'] = games_with_rating.apply(final_stars, axis=1)

In [10]:
games_with_rating.sort_values(by = 'final rating', ascending = False)

rank                                              title  total ratings  \
741     18                               超級單字王 - 英檢、多益、托福 輕鬆學         156922   
956     33                                    Indy Cat for VK         999159   
1658    31                     Word Collect - Free Word Games         406650   
813     90               Английский для Начинающих: LinDuo HD          50859   
575     54               Free Slot Machines with Bonus Games!         206215   
...    ...                                                ...            ...   
1592    69                                      NEW QuizDuel!          65544   
1606    83  TopQuiz -Play Quiz & Lottery | Win Money via P...          55050   
304      5                                              모두의마블        1596758   
1590    67                                              스케치퀴즈          65734   
873     50                 Au Mobile VTC – Game nhảy Audition         103881   

     installs  average rating  growth (30 days)  growth (60 days)  price  \
741   500.0 k            4.91               0.6               1.5    0.0   
956     5.0 M            4.87               0.0               0.0    0.0   
1658   10.0 M            4.82               3.7               9.4    0.0   
813     1.0 M            4.82               4.5               8.3    0.0   
575     5.0 M            4.82               0.3               0.6    0.0   
...       ...             ...               ...               ...    ...   
1592    1.0 M            3.21               7.4              17.8    0.0   
1606    1.0 M            3.20               0.1               0.4    0.0   
304    10.0 M            3.16               0.0               0.0    0.0   
1590    5.0 M            3.13               0.0               0.0    0.0   
873     5.0 M            3.09               0.1               0.3    0.0   

              category  5 star ratings  4 star ratings  3 star ratings  \
741   GAME EDUCATIONAL          148058            6185            1428   
956        GAME PUZZLE          932203           35588           10645   
1658         GAME WORD          373268           16566            4404   
813   GAME EDUCATIONAL           44723            4611             707   
575        GAME CASINO          188876            7861            3876   
...                ...             ...             ...             ...   
1592       GAME TRIVIA           24833           11671            3568   
1606       GAME TRIVIA           24578            4760            2971   
304         GAME BOARD          666378          138474          141241   
1590       GAME TRIVIA           25093            6939            6959   
873         GAME MUSIC           44324            6906            6996   

      2 star ratings  1 star ratings   paid  final rating  
741              369             879  False      4.912923  
956             5257           15464  False      4.865382  
1658            3184            9225  False      4.823370  
813              288             527  False      4.823089  
575             1265            4334  False      4.821814  
...              ...             ...    ...           ...  
1592            3499           21970  False      3.212050  
1606            2454           20284  False      3.197904  
304            89312          561350  False      3.162340  
1590            4763           21977  False      3.127915  
873             5536           40117  False      3.094187  

[1730 rows x 16 columns]

In [11]:
games_with_rating['final rating'].describe()

count    1730.000000
mean        4.313373
std         0.253587
min         3.094187
25%         4.180504
50%         4.332741
75%         4.490263
max         4.912923
Name: final rating, dtype: float64

There's not a lot of variation in the final ratings and that's to be expected since this is a dataset from the top 100 games of each category. 

## Highest Rated Category

We need a metric to calculate the highest rating of each category. Here we'll use the average rating from the category to check the highest.

In [12]:
games_with_rating['category'].unique()

array(['GAME ACTION', 'GAME ADVENTURE', 'GAME ARCADE', 'GAME BOARD',
       'GAME CARD', 'GAME CASINO', 'GAME CASUAL', 'GAME EDUCATIONAL',
       'GAME MUSIC', 'GAME PUZZLE', 'GAME RACING', 'GAME ROLE PLAYING',
       'GAME SIMULATION', 'GAME SPORTS', 'GAME STRATEGY', 'GAME TRIVIA',
       'GAME WORD'], dtype=object)

In [13]:
df1 = games_with_rating.groupby(['category'], as_index = False).mean()
df1[['category', 'final rating']].sort_values(by = 'final rating', ascending = False)

category  final rating
16          GAME WORD      4.442993
5         GAME CASINO      4.437586
4           GAME CARD      4.421315
9         GAME PUZZLE      4.393757
3          GAME BOARD      4.347517
6         GAME CASUAL      4.335134
12    GAME SIMULATION      4.317544
10        GAME RACING      4.312110
2         GAME ARCADE      4.295247
11  GAME ROLE PLAYING      4.275691
14      GAME STRATEGY      4.265235
1      GAME ADVENTURE      4.264376
13        GAME SPORTS      4.261386
7    GAME EDUCATIONAL      4.253470
15        GAME TRIVIA      4.245882
0         GAME ACTION      4.245805
8          GAME MUSIC      4.180331

From this, we see that the category __'GAME WORD'__ has the highest final rating of all the categories, but the values are indeed pretty close for all of them, seeing as this is a top games list.

## This is a work in progress. Adding some visualizations might be useful in the future.